src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1199,9 +1199,9 @@ def apply_fixes(
 
             before = []
             after = []
-            if fixes_applied and not r.can_start_end_non_code:
```

Comment:
Removed the `fixes_applied` check. This prevented the "bubbling up" logic from propagating whitespace multiple levels up the parse tree. In this particular issue, a newline needed to be moved **two levels** up the parse tree, but it was stopping after one, hence the critical error message from the "post-fix parse" check.